sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -203,7 +203,11 @@ object JdbcUtils extends Logging with SQLConfHelper {
     case java.sql.Types.DECIMAL | java.sql.Types.NUMERIC if scale < 0 =>
       DecimalType.bounded(precision - scale, 0)
     case java.sql.Types.DECIMAL | java.sql.Types.NUMERIC =>
-      DecimalPrecisionTypeCoercion.bounded(precision, scale)
+      DecimalPrecisionTypeCoercion.bounded(
+        precision,
+        // A safeguard in case the JDBC scale is larger than precision which is not supported by
+        // Spark.
+        math.min(scale, precision))
```

Comment:
```suggestion
      if (scale > precision) {
        // A safeguard in case the JDBC scale is larger than precision which is not supported by
        // Spark.
        DecimalPrecisionTypeCoercion.bounded(precision + scale, scale)
      } else {
        DecimalPrecisionTypeCoercion.bounded(precision, scale)
      }
```